internal/provisioner/plain/pkg/git/git.go

```diff
@@ -12,14 +14,26 @@ const (
 	defaultBranch    = "main"
 )
 
-func CheckoutCommand(source v1alpha1.GitSource) string {
+type CheckoutCmd struct {
+	v1alpha1.GitSource
+}
+
+func NewCheckoutCmd(s v1alpha1.GitSource) CheckoutCmd {
+	return CheckoutCmd{GitSource: s}
+}
+
+func (c CheckoutCmd) String() (string, error) {
 	var checkoutCommand string
-	var repository = source.Repository
-	var repositoryName = strings.Split(repository, "/")[4]
-	var directory = source.Directory
-	var branch = source.Ref.Branch
-	var commit = source.Ref.Commit
-	var tag = source.Ref.Tag
+	var repository = c.Repository
+	var directory = c.Directory
+	var branch = c.Ref.Branch
+	var commit = c.Ref.Commit
+	var tag = c.Ref.Tag
+
+	repositoryName, err := c.ValidateRepo()
+	if err != nil {
+		return "", err
+	}
```

Comment:
Can we move this kind of logic to NewCheckoutCmd constructor? The String() method should be only responsible for returning a formatted string, e.g. return `fmt.Sprintf("git checkout %s && cd %s ...", c.Repo, c.Name, ...)`.